tutorials/persistance-comparison-series/File/file.md

```diff
@@ -0,0 +1,345 @@
+
+<img src="images/08_hit_count_extended_file.jpg" alt="File hitCountExtended.txt" width=""/>
+
+As long as all the data saved to those lines belongs together, the file can be one option. If you have several different properties you might create multiple files. Alternatively you can save all the data into the same file using a bit of structure. One of the possible approaches to structuring your data will be shown in the next section.
+
+## More complex data
+
+(see `FileExampleJson.cs` in the repository for the finished version)
+
+JSON is a very common approach when saving structured data. It's easy to use and there are framework for almost any language. The .NET framework provides a [`JsonSerializer`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=net-6.0). Unity has it's own version of it: [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html)
+
+As you can see in the documentation the functionality boils down to these three methods:
+
+- *FromJson*: Create an object from its JSON representation.
+- *FromJsonOverwrite*: Overwrite data in an object by reading from its JSON representation.
+- *ToJson*: Generate a JSON representation of the public fields of an object.
+
+The `JsonUtility` transforms JSON into objects and back. Therefore our first change to the previous section is to define such an object with public fields:
+
+```cs
+private class HitCount
+{
+    public int Unmodified;
+    public int Shift;
+    public int Control;
+}
+```
+
+The class itself can be `private` and just be added inside the `FileExampleJson` class but its fields need to be public.
+
+As before we use a different file to save this data. Update the filename to:
+
+```cs
+private const string HitCountFileJson = "hitCountFileJson.txt";
+```
+
+When saving the data we will use the same `Update()` method as before to detect which key was pressed.
+
+The first part of `OnMouseDown()` (1) can stay the same as well since this part only increments the hit count in depending on the modifier used.
+
+```cs
+private void OnMouseDown()
+{
+    // 1
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift)
+    {
+        // Increment the Shift hit count.
+        hitCountShift++;
+    }
+    else if (keyPressed == KeyCode.LeftControl)
+    {
+        // Increment the Control hit count.
+        hitCountControl++;
+    }
+    else
+    {
+        // If neither Shift nor Control was held, we increment the unmodified hit count.
+        hitCountUnmodified++;
+    }
+
+    // 2
+    // Create a new HitCount object to hold this data.
+    HitCount hitCount = new();
+    hitCount.Unmodified = hitCountUnmodified;
+    hitCount.Shift = hitCountShift;
+    hitCount.Control = hitCountControl;
+
+    // 3
+    // Create a JSON using the HitCount object.
+    string jsonString = JsonUtility.ToJson(hitCount, true);
+
+    // 4
+    // Save the json to the file.
+    File.WriteAllText(HitCountFileJson, jsonString);
```

Comment:
This section is also missing the code for reading the json data and initializing the values.

Reply:
Also included in 9a5844e